src/actions/cart-manager.ts

```diff
@@ -0,0 +1,92 @@
+import {retrieveSpecificProduct} from './DAL/product-db-actions';
+
+export const mainCart: { id: string, users: { userId: string, amount: number }[] }[] = [];
+
+export const addToCart = async (ip: string, productId: string, desiredAmount: number) => {
```

Comment:
ip?

Reply:
you right...
before I used cookie I managed users by their ip